src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs

```diff
@@ -146,7 +179,8 @@ public bool TryGetHasUnscopedRefAttribute(out bool hasUnscopedRefAttribute)
 
             if (propEx != null || isBad)
             {
-                result._lazyCachedUseSiteInfo.Initialize(new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, result));
+                result.AccessUncommonFields()._lazyCachedUseSiteInfo.Initialize(new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, result));
```

Comment:
>result.AccessUncommonFields()._lazyCachedUseSiteInfo.Initialize [](http://example.com/codeflow?start=16&length=63)

`PEMethodSymbol` has a dedicated `InitializeUseSiteDiagnostic()` method that is used for all (?) cases where use-site diagnostics are calculated, and that also calls `SetUseSiteDiagnosticPopulated()`. Is there a reason t set the fields directly here rather than borrowing that approach?

Reply:
I'm frankly not entirely certain why PEMethodSymbol has as complicated an initialization strategy as it does, but it does also initialize the diagnostic from more properties than we do in PEPropertySymbol. Here, we only initialize from:

1. The creation steps, where the created symbol has not yet been exposed to anything else, so we don't have to worry about races.
2. `GetUseSiteDiagnostic`, which does have to worry about races.

The pattern used here is the pattern used by all the rest of the `UncommonFields` members, which correctly handles the occasional race of calculating that there is no use-site diagnostic/setting `_flags.IsUseSiteDiagnostic` and creating `_uncommonFields`.